File: abtraders1/public/js/sales_invoice.js
frappe.ui.form.on('Sales Invoice', { 

    customer: function (frm) {
        frappe.call({
            method: 'erpnext.accounts.utils.get_balance_on',
            args: {
                doctype: 'Sales Invoice',
                party_type: 'Customer',
                party: frm.doc.customer,
                outstanding_only: 1
            },
            callback: function (r) {
                if (r.message) {
                    frm.set_value('custom_current_outstanding', r.message);
                }
            }
        });
    },
  });

File: abtraders1/abtraders1/events/sales_invoice.py
def custom_before_save(doc, method):
     current_outstanding = float(doc.custom_current_outstanding) if doc.custom_current_outstanding else 0
     rounded_total = float(doc.rounded_total) if doc.rounded_total else 0
     doc.custom_total_with_outstanding = rounded_total + current_outstanding
